### PR TITLE
build(client): Use pnpm in client release group

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR changes the client release group to use pnpm instead of npm. lerna now uses the pnpm workspace config and uses pnpm to execute scripts.

## Notes for reviewers

There was a missing dependency that I corrected: https://github.com/microsoft/FluidFramework/pull/13554/files#r1087225147
While ideally this change would be separate from this PR, it's only the test-utils project and I'd like to avoid another CI run just for this update.

## Things that should still work

Note: `fluid-build` === `npm run build:fast`

- fluid-build should continue to build the repo incrementally.
- `fluid-build --install` should use pnpm for the release groups that use it and npm elsewhere.
- You can still use npm to run scripts, e.g. `npm run build`, or you can switch to pnpm e.g. `pnpm build`.

### CODE REVIEW

**Comment by alexvy86** [comment]
👀

File: packages/test/test-driver-definitions/tsconfig.json [hunk 1]
```diff
@@ -6,11 +6,11 @@
     "compilerOptions": {
         "rootDir": "./src",
         "outDir": "./dist",
-        "types": [ "node" ],
+        // "types": [ "node" ],
         "composite": true
```
Comment (tylerbutler) [inline packages/test/test-driver-definitions/tsconfig.json:10]:
```suggestion
        "composite": true
```

File: tools/pipelines/build-client.yml [hunk 2]
```diff
@@ -98,23 +99,3 @@ extends:
...
-        command: 'custom'
-        customCommand: 'ci --ignore-scripts'
-        customRegistry: 'useNpmrc'
-    - task: Bash@3
-      displayName: 'Generate Mono repo package json'
-      inputs:
-        targetType: 'inline'
-        script: |
-          # Generate the package/package lock for the lerna project so we would scan it.
-          node node_modules/@fluidframework/build-tools/dist/genMonoRepoPackageJson/genMonoRepoPackageJson.js
```
Comment (alexvy86) [inline tools/pipelines/build-client.yml:118]: We need to move all release groups to pnpm before getting rid of the script and its folder, right?

File: experimental/PropertyDDS/examples/property-inspector/src/theme.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import "@hig/fonts/build/ArtifaktElement.css";
 import { createTheme } from "@material-ui/core/styles";
 import { ToggleButtonClassKey } from "@material-ui/lab/ToggleButton";
 
-declare module "@material-ui/core/styles/overrides" {
+declare module "@material-ui/core/styles" {
```
Comment (alexvy86) [inline experimental/PropertyDDS/examples/property-inspector/src/theme.ts:10]: Not quite sure what the changes in this file are doing. But if they're necessary, shouldn't they also be necessary in `experimental/PropertyDDS/packages/property-inspector-table/src/Theme.tsx`?

**Comment by CraigMacomber** [comment]
I checked out this branch and acted how a normal dev might if they didn't know anything special was happening. I ran "npm i", and did as directed in the nice error message.

**Incremental install time is now ~2.8 seconds instead of ~150 seconds.**
Clean install (deleted node_modules) is now ~8.6 seconds instead of ~98 (yes, incremental npm install was slower than clean).

Big wins!

I have confirmed that my primary development workflow seems to be fine.

I did notice `api-report/container-runtime.api.md` got updated during the build: I think you need to commit that.

**Comment by CraigMacomber** [comment]
I think it might make sense to merge the package dependency fixes as one PR, then do this one as a follow up. That will make it easier to revert and un-revert this if needed, as well as simplify review of it.

**Comment by Josmithr** [comment]
My standard workflow appears to be working as well, minus the current build failure in container-loader. Installation time improvements are *definitely* noticeable. Loving this change!

File: tools/pipelines/build-client.yml (hunk 2 above)
Comment (tylerbutler) [reply tools/pipelines/build-client.yml:118]: Yeah; for any release groups that still have lerna-lockfile.json we need to keep this task.

**Comment by tylerbutler** [comment]
> I think it might make sense to merge the package dependency fixes as one PR, then do this one as a follow up. That will make it easier to revert and un-revert this if needed, as well as simplify review of it.

I will try, but the reason I didn't do this is because hoisting has in the past led to weird issues that pnpm doesn't suffer from because of its strictness. That said, I didn't actually try it, so I should.

File: experimental/PropertyDDS/examples/property-inspector/src/theme.ts (hunk 3 above)
Comment (tylerbutler) [reply experimental/PropertyDDS/examples/property-inspector/src/theme.ts:10]: Reverting these changes yields:

```
@fluid-experimental/property-inspector: error during command 'tsc'
@fluid-experimental/property-inspector: src/theme.ts:10:16 - error TS2664: Invalid module name in augmentation, module '@material-ui/core/styles/overrides' cannot be found.
@fluid-experimental/property-inspector: 
@fluid-experimental/property-inspector: 10 declare module "@material-ui/core/styles/overrides" {
@fluid-experimental/property-inspector:                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@fluid-experimental/property-inspector: 
@fluid-experimental/property-inspector: src/theme.ts:24:5 - error TS2322: Type '{ MuiButton: { root: { borderRadius: string; }; }; MuiToggleButton: { root: { "&:not(:first-child)": { borderLeft: string; marginLeft: string; }; }; }; }' is not assignable to type 'Overrides'.
@fluid-experimental/property-inspector:   Object literal may only specify known properties, but 'MuiToggleButton' does not exist in type 'Overrides'. Did you mean to write 'MuiStepButton'?
@fluid-experimental/property-inspector: 
@fluid-experimental/property-inspector:  24     MuiToggleButton: {
@fluid-experimental/property-inspector:         ~~~~~~~~~~~~~~~~~~
@fluid-experimental/property-inspector:  25       root: {
@fluid-experimental/property-inspector:     ~~~~~~~~~~~~~
@fluid-experimental/property-inspector: ... 
@fluid-experimental/property-inspector:  30       },
@fluid-experimental/property-inspector:     ~~~~~~~~
@fluid-experimental/property-inspector:  31     },
@fluid-experimental/property-inspector:     ~~~~~
@fluid-experimental/property-inspector: 
@fluid-experimental/property-inspector: 
@fluid-experimental/property-inspector: Found 2 errors.
```

**Comment by tylerbutler** [comment]
@CraigMacomber I've pulled the dependency changes into #13612. Once that's merged this PR should shrink dramatically.

**Comment by alexvy86** [comment]
I think this PR should include the update to `pnpm` in this section of the root `README.md`:

image

The wiki will probably require some updates as well? Maybe some of the internal docs?

File: experimental/PropertyDDS/packages/property-dds/tsconfig.json [hunk 4]
```diff
@@ -9,7 +9,7 @@
 		"noUnusedLocals": false,
 		"outDir": "./dist",
 		"rootDir": "./src",
-		"types": ["node"]
+        "types": ["node"]
```
Comment (CraigMacomber) [inline experimental/PropertyDDS/packages/property-dds/tsconfig.json:12]: The other lines in the file use tabs: so changing this one to spaces seems bad.

Comment (tylerbutler) [reply experimental/PropertyDDS/packages/property-dds/tsconfig.json:12]: Good catch; fixed.

**Comment by tylerbutler** [comment]
> I think this PR should include the update to `pnpm` in this section of the root `README.md`:
> 
> image
> 
> The wiki will probably require some updates as well? Maybe some of the internal docs?

I put the doc updates in a separate PR: #13729

File: packages/test/test-utils/package.json [hunk 5]
```diff
@@ -95,6 +95,7 @@
     "@fluidframework/test-utils-previous": "npm:@fluidframework/test-utils@2.0.0-internal.2.2.0",
     "@microsoft/api-extractor": "^7.22.2",
     "@rushstack/eslint-config": "^2.5.1",
+    "@types/debug": "^4.1.5",
```
Comment (tylerbutler) [inline packages/test/test-utils/package.json:98]: This missing dep must have been introduced by the next branch merge.

**Review by CraigMacomber — APPROVED** [review]

**Pull request merged**